### ISSUE TITLE
Diff only for selected paths

### ISSUE THREAD

**Issue by Vampire** [git-multimail/git-multimail#195]
The default mail hook for SVN had a nice feature.
You could specify via config file which paths you are interested in.
The diff mails then had diffs only for the selected paths and for the other paths in the same commit it just said something like "These files also have changes, but their diff is omitted.".
Would be nice if git-multimail also supports this. 
I can already restrict the mails being sent by first checking the paths and only triggering git-multimail if the push is relevant, but if it is relevant then all diffs are sent. 
And if I would configure the paths by additional diff arguments, then the non matching files are not mentioned at all.

**Comment by moy** [comment]
I agree it would be nice, but I'm afraid the effort to implement this won't fit in my time budget any time soon. Patches welcome, obviously.

**Comment by zoidbergwill** [comment]
Respecting something like GitHub's `linguist-generated=true` in `.gitattributes` could be useful too. 

It's mentioned in this blogpost: https://medium.com/@clarkbw/managing-generated-files-in-github-1f1989c09dfd

**Comment by zoidbergwill** [comment]
This overlaps with "Allow people to subscribe to changes only to particular files." in the [notes.txt](https://github.com/git-multimail/git-multimail/blob/master/notes.txt#L69-L71), so I'm gonna mention it here.

In a private fork I've made it possible to create a YAML file, that allows people to subscribe to e-mails for specific commits, based on project, author, and changed files in the commit, but I'm not sure how I could break the change into changes I could contribute upstream. It's a lot more complicated than the existing `send_emails`, but if it's something that people would be interested using, I can try do that.

Sorry if these things don't really overlap since our current solution just e-mails folks all changed files in the commit if any match their subscription options, and doesn't necessarily hide other files in the diff. If this belongs in another issue, I can do that.

I'd also like to look at supporting a GitLab webhook similar to https://github.com/git-multimail/git-multimail/issues/109 if I get a chance, since I think the changes we've made probably would be better running there, and not slowing down people's pushes.

**Comment by Vampire** [comment]
Well, I currently have that functionality in a hook that is controlled by a simple configuration file that also non-programmers can edit and only forward to multimail if the path regex matches.

With the SVN mail hook I generated the configuration file that it accepted from this simplified input file and then always forwarded to it.

So having the functionality to only mail if a path matches is nice and can be part of the fix for this, but actually for me it would be enough if you could specify the paths of which the diffs should be included and the others only mentioned like for the SVN mail hook.

Also having the filtering whether to send at all in multimail would of course be even better, then I could reduce my custom logic to again just generate the respective configuration file and maybe even cache the generated file as long as the input file does not change.
